src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -137,798 +34,233 @@ namespace JSONHelpers
+            info.m_rowsPercents = JsonArrayToNumVec(rowsPercentage);
+            info.m_columnsPercents = JsonArrayToNumVec(columnsPercentage);
+            for (const auto& cellsRow : cellChildMap)
             {
-                if (auto zoneSet = CustomZoneSetJSON::FromJson(customZoneSets.GetObjectAt(i)); zoneSet.has_value())
+                const auto cellsArray = cellsRow.GetArray();
+                if (cellsArray.Size() != info.m_columns)
                 {
-                    customZoneSetsMap[zoneSet->uuid] = std::move(zoneSet->data);
+                    return std::nullopt;
                 }
+                info.cellChildMap().push_back(JsonArrayToNumVec(cellsArray));
             }
 
-            return true;
+            return info;
         }
         catch (const winrt::hresult_error&)
         {
-            return false;
+            return std::nullopt;
         }
     }
 
-    json::JsonArray FancyZonesData::SerializeCustomZoneSets() const
+    json::JsonObject CustomZoneSetJSON::ToJson(const CustomZoneSetJSON& customZoneSet)
     {
-        std::scoped_lock lock{ dataLock };
-        json::JsonArray customZoneSetsJSON{};
+        json::JsonObject result{};
 
-        for (const auto& [zoneSetId, zoneSetData] : customZoneSetsMap)
+        result.SetNamedValue(L"uuid", json::value(customZoneSet.uuid));
+        result.SetNamedValue(L"name", json::value(customZoneSet.data.name));
+        switch (customZoneSet.data.type)
         {
-            customZoneSetsJSON.Append(CustomZoneSetJSON::ToJson(CustomZoneSetJSON{ zoneSetId, zoneSetData }));
-        }
-
-        return customZoneSetsJSON;
-    }
-
-    void FancyZonesData::LoadFancyZonesData()
-    {
-        std::scoped_lock lock{ dataLock };
-        std::wstring jsonFilePath = GetPersistFancyZonesJSONPath();
-
-        if (!std::filesystem::exists(jsonFilePath))
+        case FancyZonesDataTypes::CustomLayoutType::Canvas:
         {
-            MigrateCustomZoneSetsFromRegistry();
+            result.SetNamedValue(L"type", json::value(L"canvas"));
 
-            SaveFancyZonesData();
-        }
-        else
-        {
-            json::JsonObject fancyZonesDataJSON = GetPersistFancyZonesJSON();
+            FancyZonesDataTypes::CanvasLayoutInfo info = std::get<FancyZonesDataTypes::CanvasLayoutInfo>(customZoneSet.data.info);
+            result.SetNamedValue(L"info", CanvasLayoutInfoJSON::ToJson(info));
 
-            ParseAppZoneHistory(fancyZonesDataJSON);
-            ParseDeviceInfos(fancyZonesDataJSON);
-            ParseCustomZoneSets(fancyZonesDataJSON);
+            break;
         }
-    }
-
-    void FancyZonesData::SaveFancyZonesData() const
-    {
-        std::scoped_lock lock{ dataLock };
-        json::JsonObject root{};
-        json::JsonObject appZoneHistoryRoot{};
+        case FancyZonesDataTypes::CustomLayoutType::Grid:
+        {
+            result.SetNamedValue(L"type", json::value(L"grid"));
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.